modules/programs/kodi.nix

```
@@ -0,0 +1,238 @@
+      '';
+    };
+
+    datadir = mkOption {
+      type = types.path;
+      default = "${config.home.homeDirectory}/.kodi";
+      defaultText = literalExpression "\"\${config.home.homeDirectory}/.kodi\"";
+      description = "Directory to store configuration and metadata.";
+      example = literalExpression "\"\${config.xdg.dataHome}/kodi\"";
+    };
+
+    settings = mkOption {
+      type = with types; nullOr (attrsOf anything);
+      default = null;
+      description = ''
+        Configuration to write to the <literal>advancedsettings.xml</literal>
+        file in kodis userdata directory. Settings specified here will be
+        immutable from inside kodi and be hidden from the GUI settings dialog.
+
+        See <link xlink:href="https://kodi.wiki/view/Advancedsettings.xml"/> as
+        reference for how settings need to be specified.
+
+        The innermost attributes must be of type str.
+      '';
+      example = literalExample ''
+        { videolibrary.showemptytvshows = "true"; }
+      '';
+    };
+
+    sources = mkOption {
+      type = with types; nullOr (attrsOf anything);
+      default = null;
+      description = ''
+        Contents to populate the file <literal>sources.xml</literal> in kodis
+        userdata directory.
+
+        See <link xlink:href="https://kodi.wiki/view/Sources.xml"/> as
+        reference for how sources need to be specified.
+
+        Kodi will still show the dialogs to modify sources in the GUI and they
+        appear to be mutable. This however is not the case and the sources will
+        stay as specified via home-manager.
+      '';
+      example = literalExample ''
+        {
+          video = {
+            default = "movies";
+            source = [
+              { name = "videos"; path = "/path/to/videos"; allowsharing = "true"; }
+              { name = "movies"; path = "/path/to/movies"; allowsharing = "true"; }
+            ];
+          };
+        }
+      '';
+    };
+
+    addonSettings = mkOption {
+      type = with types; nullOr (attrsOf (attrsOf str));
+      default = null;
+      description = ''
+        Attribute set with the plugin namespace as toplevel key and the plugins
+        settings as lower level key/value pairs.
+
+        Kodi will still show the settings of plugins configured via this
+        mechanism in the GUI and they appear to be mutable. This however is
+        not the case and the settings will stay as specified via home-manager.
+      '';
+      example = literalExample ''
+        { "service.xbmc.versioncheck".versioncheck_enable = "false"; }
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ];
+      home.sessionVariables = { KODI_DATA = cfg.datadir; };
+    }
+
+    (mkIf (!isNull cfg.settings) {
+      home.file."${cfg.datadir}/userdata/advancedsettings.xml".source =
+        (attrsetToAdvancedSettingsXml cfg.settings "kodi-advancedsettings.xml");
```

Unnecessary parentheses
```suggestion
        attrsetToAdvancedSettingsXml cfg.settings "kodi-advancedsettings.xml";
```

Same below.